### ISSUE TITLE
com.hazelcast.mapstore.GenericMapStoreTest.whenMapStoreDestroyOnNonMaster_thenDropMapping [HZ-1816]

### ISSUE THREAD

**Issue by ramizdundar** [hazelcast/hazelcast#22946]
https://github.com/hazelcast/hazelcast/pull/22941#issuecomment-1326597448 (commit 47673944465b32875c60c6125ab6a2e26245947c)

Failed on `Hazelcast-pr-builder`: https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/14461/testReport/junit/com.hazelcast.mapstore/GenericMapStoreTest/whenMapStoreDestroyOnNonMaster_thenDropMapping/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting:
  <[Row{[__map-store.people_o_3a213dc2_bf2b_47a1_a1ff_6a896375bc2c]},
    Row{[__map-store.people_o_5298b7ff_cb8c_4b10_8042_87b96c91b01e]}]>
to contain exactly in any order:
  <[Row{[__map-store.people_o_3a213dc2_bf2b_47a1_a1ff_6a896375bc2c]}]>
but the following elements were unexpected:
  <[Row{[__map-store.people_o_5298b7ff_cb8c_4b10_8042_87b96c91b01e]}]>

	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:322)
	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:284)
	at com.hazelcast.mapstore.GenericMapStoreTest.lambda$awaitMappingCreated$3(GenericMapStoreTest.java:668)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1253)
	at com.hazelcast.mapstore.GenericMapStoreTest.awaitMappingCreated(GenericMapStoreTest.java:667)
	at com.hazelcast.mapstore.GenericMapStoreTest.whenMapStoreDestroyOnNonMaster_thenDropMapping(GenericMapStoreTest.java:136)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: whenSetNonExistingColumnOnSecondMapStore_thenFailToInitialize in 0.039 seconds.
Started Running Test: whenMapStoreDestroyOnNonMaster_thenDropMapping
15:29:56,960 DEBUG || - [GenericMapStore] hz.tender_borg.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing for map people_o_3a213dc2_bf2b_47a1_a1ff_6a896375bc2c
15:29:56,963 DEBUG || - [GenericMapStore] hz.tender_borg.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Discovered following mapping columns: id INTEGER, name VARCHAR, age INTEGER
15:29:56,964  WARN || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_351]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_351]
15:29:56,964  WARN || - [SubmitJobOperation] hz.tender_borg.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_351]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_351]
15:29:56,975  WARN || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_351]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_351]
15:29:56,976  WARN || - [SubmitJobOperation] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:547) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:334) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:328) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_351]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_351]
15:30:02,305  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
15:30:02,306  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
15:30:02,318  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
15:30:02,320  INFO |whenMapStoreDestroyOnNonMaster_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
BuildInfo right after whenMapStoreDestroyOnNonMaster_thenDropMapping(com.hazelcast.mapstore.GenericMapStoreTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20221124', buildNumber=20221124, revision=ada41fc, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'whenMapStoreDestroyOnNonMaster_thenDropMapping(com.hazelcast.mapstore.GenericMapStoreTest):'
15:29:55, accumulated pauses: 40 ms, max pause: 7 ms, pauses over 1000 ms: 0
15:30:00, accumulated pauses: 184 ms, max pause: 170 ms, pauses over 1000 ms: 0


No metrics recorded during the test
```

</details>

**Comment by ramizdundar** [comment]
Maybe related to https://github.com/hazelcast/hazelcast/issues/22910 cc @TomaszGaweda

**Comment by TomaszGaweda** [comment]
Very likely it's the same

**Comment by burakgok** [comment]
This issue duplicates #21993 and should have been resolved by #22806.

**Comment by TomaszGaweda** [comment]
to-jira
